log4j-api/src/main/java/org/apache/logging/log4j/message/ParameterFormatter.java

```diff
@@ -236,7 +240,10 @@ static void formatMessage(
             final String message = String.format(
                     "found %d argument placeholders, but provided %d for pattern `%s`",
                     analysis.placeholderCount, args.length, pattern);
-            throw new IllegalArgumentException(message);
```

Comment:
Personally I am pretty much in favor of throwing an exception here: [`java.util.Formatter`](https://docs.oracle.com/javase/8/docs/api/java/util/Formatter.html) does it, so we can also throw.

On the other hand we should probably catch the exception in `ParameterizedMessage#getFormattedMessage`, as it happens in other kinds of messages:

https://github.com/apache/logging-log4j2/blob/28b70d1d7fe63417e8eaa1abe8bf992363060515/log4j-api/src/main/java/org/apache/logging/log4j/message/StringFormattedMessage.java#L120-L126

https://github.com/apache/logging-log4j2/blob/28b70d1d7fe63417e8eaa1abe8bf992363060515/log4j-api/src/main/java/org/apache/logging/log4j/message/MessageFormatMessage.java#L113-L119

Reply:
I have updated it as you requested. Could you review the changes in 1084855467f554f2dabf34af4d860018fb098fd8, please?